tests/e2e/provisioning/internal/hyperscaler/account_pool.go

```diff
@@ -0,0 +1,105 @@
+package hyperscaler
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type HyperscalerType string
```

Comment:
(nit) Hyperscaler is redundant, it reads `hyperscaler.HyperscalerType`.

Reply:
Changed to `Type`